File: src/com/divisiblebyzero/ada/player/ComputerPlayer.java
package com.divisiblebyzero.ada.player;

//
// ada.player.ComputerPlayer.java
// Ada Chess
//
// Created by Eric Czarny on August 21, 2009.
// Copyright 2009 Divisible by Zero. All rights reserved.
//

import com.divisiblebyzero.ada.view.component.Board;
import com.divisiblebyzero.chess.Move;
import com.divisiblebyzero.chess.ai.Search;

public class ComputerPlayer implements Player {
    
    public void makeMove(Board board, int color) {
        SearchThread searchThread = new SearchThread(board, color);
        
        searchThread.start();
    }
    
    private class SearchThread extends Thread {
        private Board board;
        private int color;
        
        public SearchThread(Board board, int color) {
            this.board = board;
            this.color = color;
        }
        
        public void run() {
            Search search = new Search(3);
            Move move = search.searchForMove(this.board.getBitmaps(), this.color);
            
            this.board.makeMove(move);
        }
    }
}

File: src/main/scala/com/divisiblebyzero/ada/player/ComputerPlayer.scala
package com.divisiblebyzero.ada.player

import com.divisiblebyzero.ada.view.component.Board
import com.divisiblebyzero.chess.Move
import com.divisiblebyzero.chess.ai.Search

class ComputerPlayer extends Player {
  class SearchThread extends Thread {
    def this(board: Board, color: Int) {
      this ()

      this.board = board
      this.color = color
    }

    override def run() {
      val search: Search = new Search
      val move: Move = search.searchForMove(board.getBitboard, color)

      board.makeMove(move)
    }

    private var board: Board = null
    private var color: Int = 0
  }

  override def makeMove(board: Board, color: Int) {
    val searchThread = new SearchThread(board, color)

    searchThread.start()
  }
}
